core/src/main/java/com/linecorp/armeria/client/AbstractClientFactory.java

```diff
@@ -96,18 +96,17 @@
     protected final Scheme validateScheme(URI uri) {
         requireNonNull(uri, "uri");
 
-        final String scheme = uri.getScheme();
-        if (scheme == null) {
-            throw new IllegalArgumentException("URI with missing scheme: " + uri);
-        }
-
         if (uri.getAuthority() == null) {
             throw new IllegalArgumentException("URI with missing authority: " + uri);
         }
 
+        final String scheme = uri.getScheme();
+        if (scheme == null) {
```

Comment:
Is there any reason that you changed the order of checking `null`?

Reply:
To group processing. We only check and work on value of getAuthority() once. Rest of the code is related to uri scheme. So I updated the order, to have all scheme related processing grouped together. More for readability.

Reply:
Yeah, that makes sense. ;)